File: packages/java-sdk/src/main/java/io/schlepengine/exceptions/ConfigurationException.java
package io.schlepengine.exceptions;

/**
 * Exception thrown when there's a configuration error with the Schlep-engine client.
 *
 * <p>This typically occurs when:</p>
 * <ul>
 *   <li>The API key is missing or invalid</li>
 *   <li>The base URL is malformed</li>
 *   <li>Required environment variables are not set</li>
 * </ul>
 *
 * @author Schlep-engine Team
 * @version 1.0.0
 * @since 1.0.0
 */
public class ConfigurationException extends Exception {

    /**
     * Create a new ConfigurationException with the given message.
     *
     * @param message The error message describing the configuration issue
     */
    public ConfigurationException(String message) {
        super(message);
    }

    /**
     * Create a new ConfigurationException with the given message and cause.
     *
     * @param message The error message describing the configuration issue
     * @param cause The underlying cause of the configuration error
     */
    public ConfigurationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: packages/csharp-sdk/Exceptions/ConfigurationException.cs
namespace SchlepEngine.Exceptions;

/// <summary>
/// Exception thrown when there's a configuration error with the Schlep-engine client.
/// </summary>
/// <remarks>
/// This typically occurs when:
/// <list type="bullet">
/// <item>The API key is missing or invalid</item>
/// <item>The base URL is malformed</item>
/// <item>Required environment variables are not set</item>
/// </list>
/// </remarks>
public class ConfigurationException : Exception
{
    /// <summary>
    /// Initializes a new instance of the <see cref="ConfigurationException"/> class with the given message.
    /// </summary>
    /// <param name="message">The error message describing the configuration issue.</param>
    public ConfigurationException(string message) : base(message)
    {
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="ConfigurationException"/> class with the given message and inner exception.
    /// </summary>
    /// <param name="message">The error message describing the configuration issue.</param>
    /// <param name="innerException">The exception that is the cause of the current exception.</param>
    public ConfigurationException(string message, Exception innerException) : base(message, innerException)
    {
    }
}